Try to support Vite application and shared shadcn/ui components.

evgenius1424 opened the pull request:
I've added vite and copied some boilerplate. But it fails with

```
../../packages/ui/src/components/ui/button.tsx:5:20 - error TS2307: Cannot find module '@ui/lib/utils' or its corresponding type declarations.

5 import { cn } from "@ui/lib/utils";
```

If you have any ideas on how to fix it, please add comments to the review.

evgenius1424 committed:
Add vite

evgenius1424 committed:
feat: Add dependencies and configure build for Vite app

dan5py commented on apps/vite/src/App.tsx:
```diff
@@ -0,0 +1,36 @@
+import { useState } from "react";
+import reactLogo from "./assets/react.svg";
+import viteLogo from "/vite.svg";
+import "./App.css";
```
Delete the css import

dan5py commented on apps/vite/src/index.css:
Delete this file

dan5py commented on apps/vite/src/main.tsx:
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+import ReactDOM from 'react-dom/client'
+import App from './App.tsx'
+import './index.css'
```
Change the css import to:
```tsx
import '@repo/ui/globals.css'
```

dan5py commented on apps/vite/vite.config.ts:
```diff
@@ -0,0 +1,7 @@
+import { defineConfig } from 'vite'
```
Add path alias:

```ts
import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'
import path from 'node:path'

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [react()],
  resolve: {
    alias: {
      "@ui": path.resolve(__dirname, "../../packages/ui/src")
    }
  }
})
```

evgenius1424 replied:
Thanks, if was really helpful.

dan5py commented on apps/vite/postcss.config.js:
You may also need to change the file extension to `.cjs`

evgenius1424 replied:
Yep, it works.

dan5py commented on apps/vite/src/App.css:
Remove this file

evgenius1424 replied:
I've removed all vite defaults.

dan5py reviewed (commented):
Hi @evgenius1424, try applying the changes and it should work. Let me know 💪

evgenius1424 committed:
fix(Vite application) all issues resolved and now we can use shared components from ui package.

evgenius1424 commented:
Hello @dan5py, I really appreciate your help with Vite support and shared components. I wanted to start some monorepo but the shadcn-ui components only worked as a local setup.

Thanks for unblocking me and possibly others.

It works great.

<img width="697" alt="image" src="https://github.com/dan5py/turborepo-shadcn-ui/assets/13147245/276ec2e1-12df-45fa-8832-e1882f8cfea7">

evgenius1424 committed:
chore: Update README.md

Add information about sharing shadcn/ui components between nextjs and vite applications.

evgenius1424 commented:
Hi @dan5py, are you planning to merge it? I think it would be a really great update for this repo to show that even different projects can share shadcn/ui components.

starkfire commented:
+1 to this PR by @evgenius1424. I was trying to implement this template on a Turborepo project with a React.js + Vite app and I used this PR as a reference for solving the runtime issues I had when importing `@ui/lib/utils` (mentioned in #20).

I would just like to add though that the `tsconfig.json` within any Vite app should contain a path that points to the `ui` package itself.

```json
"compilerOptions": {
  "paths": {
    "@ui/*": ["../../packages/ui/src/*"]
  }
}
```

This will also solve issues where the editor may complain about not being able to find the modules being imported.

magoz commented:
Thank you both @evgenius1424  and @starkfire . Your combined solution works like a charm!

dan5py commented:
> Hi @dan5py, are you planning to merge it? I think it would be a really great update for this repo to show that even different projects can share shadcn/ui components.

@evgenius1424, actually I like the idea of having a clean repo with a single app as boilerplate. But yeah, this would help people with a different setup. I could create a simple docs website for it. I'll see what I can do.

evgenius1424 commented:
Okay, I think there are projects that use turborepo with vite only. Or do some sort of migration.

If we support two major players (VIte and Next) and point out that sharing works perfectly between them, it makes a lot of sense.

rm -rf apps/vite is all you need to continue working with next only.

But yeah, it's up to you.
